Source/WebKit/UIProcess/ProvisionalPageProxy.cpp

```diff
@@ -153,7 +153,7 @@ void ProvisionalPageProxy::cancel()
     FrameInfoData frameInfo {
         true, // isMainFrame
         m_request,
-        SecurityOriginData::fromURL(m_request.url()),
+        page().urlSchemeHandlerForScheme(m_request.url().protocol().toStringWithoutCopying()) ? SecurityOriginData::fromURLIgnoringOpaqueness(m_request.url()) : SecurityOriginData::fromURL(m_request.url()),
```

Comment:
Why can't we use `SecurityOriginData::fromURLIgnoringOpaqueness()` unconditionally here?

Reply:
I discussed this with Alex, and he made the argument that looking at the current registered custom schemes (when it's available outside of the web process) was safer than unconditionally ignoring the opaqueness checks. This is based on the observation that the primary reason why calling `fromURL` in the UIProcess is because of the difference in registered scheme handlers within the LegacySchemeRegistry.

Reply:
But we were never doing an opaqueness check here so you're changing behavior. The new code is more complicated and the change is risky.